src/webgpu/api/validation/copyTextureToTexture.spec.ts

```diff
@@ -210,3 +236,222 @@ g.test('multisampled_copy_restrictions')
       isSuccess
     );
   });
+
+g.test('uncompressed_texture_format_equality')
+  .params(
+    params()
+      .combine(poptions('srcFormat', kUncompressedTextureFormats))
+      .combine(poptions('dstFormat', kUncompressedTextureFormats))
+  )
+  .fn(async t => {
+    const { srcFormat, dstFormat } = t.params;
+
+    const kTextureSize = { width: 16, height: 16, depth: 1 };
+
+    const srcTexture = t.device.createTexture({
+      size: kTextureSize,
+      format: srcFormat,
+      usage: GPUTextureUsage.COPY_SRC,
+    });
+
+    const dstTexture = t.device.createTexture({
+      size: kTextureSize,
+      format: dstFormat,
+      usage: GPUTextureUsage.COPY_DST,
+    });
+
+    const isSuccess = srcFormat === dstFormat;
+    t.TestCopyTextureToTexture(
+      { texture: srcTexture },
+      { texture: dstTexture },
+      kTextureSize,
+      isSuccess
+    );
+  });
+
+g.test('depth_stencil_copy_restrictions')
+  .params(
+    params()
+      .combine(poptions('format', kDepthStencilFormats))
+      .combine(
+        poptions('srcCopyOrigin', [
+          { x: 0, y: 0, z: 0 },
+          { x: 1, y: 0, z: 0 },
+          { x: 0, y: 1, z: 0 },
+          { x: 1, y: 1, z: 0 },
+        ])
+      )
+      .combine(
+        poptions('dstCopyOrigin', [
+          { x: 0, y: 0, z: 0 },
+          { x: 1, y: 0, z: 0 },
+          { x: 0, y: 1, z: 0 },
+          { x: 1, y: 1, z: 0 },
+        ])
+      )
+      .combine(poptions('srcTextureSize', [64, 32]))
```

Comment:
I just realized that we do a lot of tests with the same size in X and Y, we should probably make rectangular texture so we check we correctly test width with X and height with Y.

Reply:
Done